buildbot-conf/cordova.conf

```diff
@@ -0,0 +1,336 @@
+def DescribedStep(step_class, description, haltOnFailure=True, **kwargs):
+    return step_class(description=description, descriptionDone=description, name=slugify(description), haltOnFailure=haltOnFailure, **kwargs)
+
+def SH(workdir=BASE_WORKDIR, timeout=medic_config['app']['timeout'], **kwargs):
+    return DescribedStep(ShellCommand, workdir=workdir, timeout=timeout, **kwargs)
+
+def NPM(npm_command, command=list(), what='code', **kwargs):
+    return SH(command=['npm', npm_command] + command, description='npm ' + npm_command + 'ing ' + what, **kwargs)
+
+def NPMInstall(command=list(), **kwargs):
+    # NOTE:
+    #      adding the --cache parameter so that we don't use the global
+    #      npm cache, which is shared with other processes
+    return NPM('install', command=command + [I('--cache=%(prop:workdir)s/npm_cache')], **kwargs)
+
+def NPMTest(**kwargs):
+    return NPM('test', **kwargs)
+
+def BuildbotClone(repourl, what='code', workdir=None, **kwargs):
+    if workdir is None:
+        workdir = what
+    return DescribedStep(Git, 'cloning ' + what, repourl=repourl, workdir=workdir, mode='full', method='clobber', shallow=True, **kwargs)
+
+def CordovaClone(project_name, **kwargs):
+    branch   = repo_branch_from_name(project_name)
+    repo_url = repo_url_from_name(project_name)
+    return BuildbotClone(repourl=repo_url, branch=branch, what=project_name, **kwargs)
+
+def MedicClone(config_path, excludes=list(), **kwargs):
+    """
+    Clone repositories using medic's checkout.js script.
+    """
+    command     = ['node', 'cordova-medic/bin/checkout.js', '--config=' + config_path]
+    description = 'cloning configured repos'
+
+    if len(excludes) > 0:
+        excludes_string = ','.join(excludes)
+        command        += ['--exclude', excludes_string]
+        description    += ', except ' + excludes_string
+
+    return SH(command=command, description=description)
+
+def RMRF(path, description=None, **kwargs):
+
+    if description is None:
+        description = 'removing ' + path
+
+    # this is better than 'rm -rf' because it removes excessively long paths on Windows
+    js_script = "var s = require('shelljs'); console.log('removing {path}'); s.rm('-rf', '{path}');".format(path=path)
+
+    return SH(command=['node', '-e', js_script], description=description, **kwargs)
+
+def Set(name, value, **kwargs):
+    return DescribedStep(SetProperty, 'setting ' + name, property=name, value=value, **kwargs)
+
+def Download(mastersrc, slavedest):
+    return FileDownload(mastersrc=mastersrc, slavedest=slavedest, workdir=BASE_WORKDIR)
+
+####### SLAVES
+
+# NOTE:
+#      these slave names refer to the ones specified in master.cfg,
+#      and they must remain defined in master.cfg in order to work
+#      with the master.cfg used on ci.apache.org's Buildbot
+MASTER_OSX_SLAVES     = ['cordova-ios-slave']
+MASTER_WINDOWS_SLAVES = ['cordova-windows-slave']
+
+OSX_SLAVES     = MASTER_OSX_SLAVES
+WINDOWS_SLAVES = MASTER_WINDOWS_SLAVES
```

Comment:
If we define slaves in `cordova.conf`, it is helpful to keep the other variable. It is there for reminding which slaves are declared in Apache's `master.cfg`.